SCClassLibrary/Common/Control/ServerStatus.sc

```diff
@@ -260,4 +222,68 @@ ServerStatusWatcher {
 		}
 	}
 
+	// final actions needed to finish booting
+	prFinalizeBoot {
+		ServerBoot.run(server);
+		server.sync;
+		server.initTree;
+	}
+
+	prSendNotifyRequest { |flag = true, addingStatusWatcher|
+		var doneOSCFunc, failOSCFunc;
+
+		if(hasBooted.not) { ^this };
+
+		// set up oscfuncs for possible server responses, \done or \failed
+		doneOSCFunc = OSCFunc({ |msg|
+			var newClientID = msg[2], newMaxLogins = msg[3];
+			failOSCFunc.free;
+
+			if(newClientID.notNil) {
+				// notify on:
+				// on registering scsynth sends back a free clientID and maxLogins
+				// this method doesn't fork/wait so we're still in the clear
+
+				// turn notified off (if it was on) to allow setting clientID
+				notified = false;
+				server.prHandleClientLoginInfoFromServer(newClientID, newMaxLogins);
+
+				// XXX: this is a workaround because using `serverBooting` is not reliable
+				// when server is rebooted quickly.
+				if(addingStatusWatcher) {
+					// this needs to be forked so that ServerBoot and ServerTree will definitely run before
+					// notified is true.
+					forkIfNeeded {
```

Comment:
I think this function needs to be scheduled on the AppClock like it was previously.